packages/driver/src/cy/chai/inspect.ts

```diff
@@ -124,6 +136,14 @@ export function create (chai) {
       }
     }
 
+    if (isShadowRoot(value)) {
```

Comment:
When we start recursively inspecting objects, listing out their properties, this can include elements with references to objects like `window` and `document`. And in Chrome 99, when we iterate over their properties, we end up with native StyleSheets, and from there some of their properties which are `Proxy` instances, which throw an error when we try to iterate over them.

Ultimately though, there's no reason we should be recursively enumerating the properties of a `Document`. In the same way we already abort for HTMLElements above, I added an early exit for ShadowRoot and Document, returning a string of their HTML rather than recursively enumerating their properties.